pkg/provisioner/ironic/ironic.go

```diff
@@ -651,8 +651,8 @@ func (p *ironicProvisioner) InspectHardware(force bool) (result provisioner.Resu
 	}
 
 	status, err := introspection.GetIntrospectionStatus(p.inspector, ironicNode.UUID).Extract()
-	if err != nil {
-		if _, isNotFound := err.(gophercloud.ErrDefault404); isNotFound {
+	if err != nil || force {
```

Comment:
Given that the existing flag doesn't automatically do what you want, it would make more sense to me to add a new flag (`refresh` or something) rather than try to overload the meaning.

Reply:
Thanks @zaneb . Will add it.